File: Insertion_Sort/Python/Insertion.py
def insertion(array):
	arr = len(array)
	
	for i in range(1, arr):
		key = array[i]
		j = i - 1
		while j >= 0 and key < array[j]:
			array[j+1] = array[j]
			j -= 1
		array[j + 1] = key

def print(array):
	arr = len(array)
	
	print("[")
	for i in range(arr):
		if i == arr - 1:
			print(array[i] + "]")
		else:
			print(array[i] + ", ")

File: Insertion_Sort/JAVA/Insertion.java
import java.util.*;

public class Insertion extends Sorting {
	public void sort(int array []) {
		int arr = array.length;
		for (int i = 1; i < arr; ++i) {
			int key = array[i];
			int j = i - 1;

			while (j >= 0 && array[j] > key) {
				array[j + 1] = array[j];
				j = j - 1;
			}
			array[j + 1] = key;
		}
	}
}